wp-graphql-custom-post-type-ui.php

```diff
@@ -1,234 +1,357 @@
-		$args['graphql_single_name' ] = ! empty( $type['graphql_single_name'] ) ?  $type['graphql_single_name'] : null;
-		$args['graphql_plural_name' ] = ! empty( $type['graphql_plural_name'] ) ?  $type['graphql_plural_name'] : null;
+	 * Set defaults for post type and taxonomy settings
+	 *
+	 * @param array  $args The args for the registry
+	 * @param string $name The name of the type
+	 * @param array  $type The array that composes the Type
+	 *
+	 * @return array
+	 */
+	public function add_graphql_settings_to_registry( array $args, string $name, array $type ): array {
+
+		if ( ! isset( $type['show_in_graphql'] ) || true !== (bool) $type['show_in_graphql'] ) {
+			return $args;
+		}
+
+		if ( ! isset( $type['graphql_plural_name'] ) || empty( $type['graphql_plural_name'] ) ) {
+			graphql_debug( sprintf( __( 'The graphql_plural_name is empty for the "%s" Post Type or Taxonomy.' ), $type['name'] ) );
+			return $args;
+		}
+
+		if ( ! isset( $type['graphql_single_name'] ) || empty( $type['graphql_single_name'] ) ) {
+			graphql_debug( sprintf( __( 'The graphql_single_name is empty for the "%s" Post Type or Taxonomy.' ), $type['name'] ) );
+			return $args;
+		}
+
+
+		$args['show_in_graphql']     = isset( $type['show_in_graphql'] ) ? (bool) $type['show_in_graphql'] : false;
+		$args['graphql_single_name'] = ! empty( $type['graphql_single_name'] ) ? $type['graphql_single_name'] : null;
+		$args['graphql_plural_name'] = ! empty( $type['graphql_plural_name'] ) ? $type['graphql_plural_name'] : null;
+
 		return $args;
 	}
 
 	/**
-	* Capture post type settings from form submission for saving
-	*/
-	public function before_update_post_type( $data ) {
-		$this->show_in_graphql = isset( $data[ 'cpt_custom_post_type' ]['show_in_graphql'] ) ? $data[ 'cpt_custom_post_type' ]['show_in_graphql'] : false;
-		$this->graphql_single_name = isset( $data[ 'cpt_custom_post_type' ]['graphql_single_name'] ) ? $data[ 'cpt_custom_post_type' ]['graphql_single_name'] : '';
-		$this->graphql_plural_name = isset( $data[ 'cpt_custom_post_type' ]['graphql_plural_name'] ) ? $data[ 'cpt_custom_post_type' ]['graphql_plural_name'] : '';
+	 * Capture post type settings from form submission for saving
+	 *
+	 * @param array $data
+	 */
+	public function before_update_post_type( array $data ) {
+
+		if ( ! isset( $data['cpt_custom_post_type']['graphql_single_name'] ) || empty( $data['cpt_custom_post_type']['graphql_single_name'] ) ) {
+			$this->show_in_graphql = false;
+		}
+
+		if ( ! isset( $data['cpt_custom_post_type']['graphql_plural_name'] ) || empty( $data['cpt_custom_post_type']['graphql_plural_name'] ) ) {
+			$this->show_in_graphql = false;
+		}
+
+		if ( ! isset( $data['cpt_custom_post_type']['show_in_graphql'] ) || empty( $data['cpt_custom_post_type']['show_in_graphql'] ) ) {
+			$this->show_in_graphql = false;
+		}
+
+		$this->show_in_graphql     = ! empty( $data['cpt_custom_post_type']['show_in_graphql'] ) ? $data['cpt_custom_post_type']['show_in_graphql'] : false;
```

Comment:
These !isset() and empty() tests are confusing.  And won't $this->show_in_graphql conditions above become irrelevant because of this $this->show_in_graphql = .... on line 122?

Reply:
reading this back through, I feel like I must have had other intentions that I didn't come back to 🤔 

Let me step back through.

Reply:
@markkelnar I pushed up another commit fixing this. And ya, I was definitely in the middle of something and went a different direction and didn't come back to this. 

Thanks for the detailed review. Love it! 🙌